wagtail/documents/rich_text/__init__.py

```diff
@@ -21,3 +21,7 @@ def expand_db_attributes(cls, attrs):
             return '<a href="%s">' % escape(doc.url)
         except (ObjectDoesNotExist, KeyError):
             return "<a>"
+
+    @classmethod
+    def extract_references(cls, attrs):
+        yield cls.get_model(), attrs["id"], "", ""
```

Comment:
was mentioning this earlier, but this would be a great to add type annotations to this, and perhaps use a TypedDict as it takes some cognitive load get this

Reply:
Have added some supporting comments in 0c2dee315c931a85dd577c088088aa68542ffaf1 (but the definitive description of the data format is still on the docstring for `ReferenceIndex._extract_references_from_object`, as I think it's best to avoid repeating this too much)

Reply:
🌟